File: src/main/scala/org/nygenome/als/graphdb/value/UniProtDrugCarrier.scala
package org.nygenome.als.graphdb.value

import org.apache.commons.csv.CSVRecord
import scala.collection.JavaConverters._

case class UniProtDrugModel(id:String,name:String, geneName:String,
                            genbankProteinId:String, genbankGeneId:String,
                            uniprotId:String,  uniprotTitle:String, pdbId:String,
                            geneCardId:String, geneAtlasId:String, hgncId:String,
                            species:String, drugIdList:java.util.List[String]
                           ) {

}
object UniProtDrugModel extends ValueTrait {

  def parseCSVRecord(record:CSVRecord): UniProtDrugModel = {
      UniProtDrugModel(record.get("ID"),  record.get("Name"),
        record.get("Gene Name"), record.get("GenBank Protein ID"),
        record.get("GenBank Gene ID"),record.get("UniProt ID"),
        record.get("Uniprot Title"), record.get("PDB ID"),
        record.get("GeneCard ID"), record.get("GenAtlas ID"),
        record.get("HGNC ID"), record.get("Species"),
        parseStringOnSemiColonFunction.apply(record.get("Drug IDs")).asJava
      )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/UniProtDrugCarrier.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class UniProtDrugCarrier (
        val drugModelType: String,
        val id:String,
        val name:String,
        val geneName:String,
        val genbankProteinId:String,
        val genbankGeneId:String,
        val uniprotId:String,
        val uniprotTitle:String,
        val pdbId:String,
        val geneCardId:String,
        val geneAtlasId:String,
        val hgncId:String,
        val species:String,
        val drugIdList: List<String>
){
    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): UniProtDrugCarrier =
            UniProtDrugCarrier(
                    "DRUG_CARRIER",
                    record.get("ID"),
                    record.get("Name"),
                    record.get("Gene Name"),
                    record.get("GenBank Protein ID"),
                    record.get("GenBank Gene ID"),
                    record.get("UniProt ID"),
                    record.get("Uniprot Title"),
                    record.get("PDB ID"),
                    record.get("GeneCard ID"),
                    record.get("GenAtlas ID"),
                    record.get("HGNC ID"),
                    record.get("Species"),
                    parseStringOnSemiColon(record.get("Drug IDs"))
            )
    }
}